test/e2e/provider_expectations.go

```diff
@@ -198,3 +199,119 @@ MachineLoop:
 	}
 	return nil
 }
+
+func (tc *testConfig) ExpectNodeToBeDrainedBeforeDeletingMachine() error {
+	listOptions := client.ListOptions{
+		Namespace: namespace,
+	}
+
+	var machine capiv1alpha1.Machine
+	var nodeName string
+	var node *corev1.Node
+
+	glog.Info("Get machineList with at least one machine with NodeRef set")
+	if err := wait.PollImmediate(1*time.Second, waitShort, func() (bool, error) {
+		machineList := capiv1alpha1.MachineList{}
+		if err := tc.client.List(context.TODO(), &listOptions, &machineList); err != nil {
+			glog.Errorf("error querying api for machineList object: %v, retrying...", err)
+			return false, nil
+		}
+		for _, machineItem := range machineList.Items {
+			// empty or non-worker role skipped
+			if machineItem.Labels["sigs.k8s.io/cluster-api-machine-role"] == "worker" {
+				if machineItem.Status.NodeRef.Name != "" {
+					machine = machineItem
+					nodeName = machineItem.Status.NodeRef.Name
+					return true, nil
+				}
+			}
+		}
+		return false, fmt.Errorf("No machine found with NodeRef not set")
+	}); err != nil {
+		return err
+	}
+
+	glog.Info("Get nodeList")
```

Comment:
why do you need to get the node? if there's actually a reason you couldn't you just fetch it right away instead of get the list?

Reply:
Just following the same pattern as is in the previous tests.